File: Data Structures & Algorithms/combination-target-sum/submission-0.cpp
class Solution {
    void dfs(int i,vector<int>temp,int sum,vector<int>&nums,int target,vector<vector<int>>& ans){
        if(sum==target){
            ans.push_back(temp);
            return;
        }

        for(int j=i;j<nums.size();j++){
            if(sum + nums[j]>target) return;

            temp.push_back(nums[j]);
            dfs(j,temp,sum+nums[j],nums,target,ans);
            temp.pop_back();
        }
    }
public:
    vector<vector<int>> combinationSum(vector<int>& nums, int target) {
        vector<vector<int>>ans;
        sort(nums.begin(),nums.end());
        dfs(0,{},0,nums,target,ans);
        return ans;
    }
};

File: Data Structures & Algorithms/subsets/submission-0.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        ans=[]
        n=len(nums)
        for i in range(1<<n):
            temp=[]
            for j in range(len(nums)):
                if i&(1<<j):
                    temp.append(nums[j])
            ans.append(temp)
        
        return ans

